### PR TITLE
Add chart for ScalarFlow Data Population

### PR DESCRIPTION
**Pull request by thongdk8**
This PR adds the helm chart for ScalarFlow data population tool, when deploying the chart, we can simply specify the command arguments for the tool by passing it to `args` value. 
This chart separates from the API chart is because we have to run this tool after the schema loader is finished and the ScalarDB server is restarted to clean the cache of updated table metadata. And also we want to run it separately sometimes.
PTAL. Thank you.

### CODE REVIEW

**Review by kota2and3kan — COMMENTED** [review]
@thongdk8  
Thank you for creating a new chart!

I'm looking at this PR, but I want to know `what scalarflow-data-population is` to review this helm chart.

It seems that there is no document (README) for `scalarflow-data-population` in the scalarflow-api repository. So, I looked at the source codes under the following two directories.

https://github.com/scalar-labs/scalarflow-api/tree/main/data-population-tool/src/main/java/com/scalar/workflow/datapopulation

https://github.com/scalar-labs/scalarflow-api/tree/main/core/src/main/java/com/scalar/workflow/core/datapopulation

As a result, it seems that the `scalarflow-data-population` loads (puts) some data for ScalarFlow into the backend DB via ScalarDB Server.

Is my understanding correct?

Also, if this tool does other tasks, could you please explain them for me to review this PR?
(It's OK to explain them briefly in bullet points. First, I want to know the overview of this tool.)

**Comment by thongdk8** [comment]
@kota2and3kan Thank you for checking the PR, I'm sorry that we haven't provided the document regarding the data-population yet, so I will briefly explain it below:
- It is a CLI tool, we pass such as population id, version to run the tool
- It is used for migrating/populating data of the tables in ScalarFlow's database, and currently via the ScalarDB server
- This tool will be run when we make changes in the table schemas such as adding a new column and requiring fill-in data for that column either from other tables or the default value.
- Whenever we run a population and it is finished, we also create a record of it as proof as it was run and insert it in the `data-population` table as well, it ensures that a population with the same id dont run twice by checking at runtime.

So in short, your understanding is correct and basically, this tool is mainly used for data migration tasks, and run separately whenever we want.

**Comment by kota2and3kan** [comment]
@thongdk8  
Thank you for your explanation! 
With those in mind, I will check this PR.

**Review by ypeckstadt — APPROVED** [review]
LGTM. Thank you.

Better to have done the ignore file in a separate PR though. It's not related to this PR.

File: scalarflow/charts/data-population/values.yaml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+  # This will clean the data population job after the job is finished (either completed or failed) after a set number of seconds. "0" means disable this feature (the job is not cleaned).
+  ttlSecondsAfterFinished: 60
+  # Arguments for the scalarflow data population
+  args: []
+  
+  scalardb:
+    config: scalardb.properties
+    configData: ""
+    # In the case the properties file is not used, then the configuration can be done via bellow configs
+    namespace: scalarflow
```
Comment (kota2and3kan) [inline scalarflow/charts/data-population/values.yaml:19]: It seems that this value is not used in any resources.
Can we remove this value?
Or, do we have to apply this value to any resources?

File: scalarflow/charts/data-population/values.yaml [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    # If you are not using ScalarDB server, you can leave this value to be empty to use the default consensus commit transaction manager
+    transactionManager: grpc
+
+  configMouthPath: /scalarflow
+
+imagePullSecrets:
+  - name: regcred
+
+nameOverride: ""
+fullnameOverride: ""
```
Comment (kota2and3kan) [inline scalarflow/charts/data-population/values.yaml:34]: Just a question. Do you have any use cases of `nameOverride` and `fullnameOverride`?
It seems that the ScalarFlow API chart and ScalarFlow Web chart do not support these two parameters.
So, I want to know the reason why the only Data Population chart supports these parameters.

File: scalarflow/charts/data-population/values.yaml [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+  # We usually recommend not to specify default resources and to leave this as a conscious
+  # choice for the user. This also increases chances charts run on environments with little
+  # resources, such as Minikube. If you do want to specify resources, uncomment the following
+  # lines, adjust them as necessary, and remove the curly braces after 'resources:'.
+  # limits:
+  #   cpu: 100m
+  #   memory: 128Mi
+  # requests:
+  #   cpu: 100m
+  #   memory: 128Mi
```
Comment (kota2and3kan) [inline scalarflow/charts/data-population/values.yaml:68]: There is no configuration `spec.template.spec.containers[].resources` in the Job resource manifest.
So, it seems that these values are not used in this chart.
Can we remove these values?
Or, do we have to apply these values to the Job resource?

**Review by kota2and3kan — COMMENTED** [review]
Thank you for creating a new chart! (And, sorry for my late review...)
I left several comments and questions.
Please take a look when you have time!

File: scalarflow/charts/data-population/templates/serviceaccount.yaml [hunk 4]
```diff
@@ -0,0 +1,12 @@
+{{- if .Values.serviceAccount.create -}}
```
[the remark this answers is not in the document]
Comment (thongdk8) [reply scalarflow/charts/data-population/templates/serviceaccount.yaml]: We dont use service account for any specific purpose, so it can be removed. Thanks for the suggestion

File: scalarflow/charts/data-population/values.yaml (hunk 1 above)
Comment (thongdk8) [reply scalarflow/charts/data-population/values.yaml:19]: It is missed from the job spec. I updated it. Thank you for pointing it out

File: scalarflow/charts/data-population/values.yaml (hunk 2 above)
Comment (thongdk8) [reply scalarflow/charts/data-population/values.yaml]: These values come from the default template, we can remove them I think. I updated to remove them

File: scalarflow/charts/data-population/values.yaml (hunk 3 above)
Comment (thongdk8) [reply scalarflow/charts/data-population/values.yaml]: It is redundant, I removed it. Thank you

**Comment by thongdk8** [comment]
@kota2and3kan Thank you for checking and leaving suggestions. I updated the chart based on your comments. PLTAL

**Review by kota2and3kan — APPROVED** [review]
LGTM! Thank you!
